Exception in API < 20 when loading plus vector

- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of FABsMenu.

###### Device/library info:
 - FABsMenu version: `1.1.1`
 - Device OS version: `API 19`
 - Device Manufacturer: `Android Virtual Device`
 - Device Name: `Nexus One`

###### Reproduction Steps

1. Open app with lib in API 19 or lower
2. Stack trace can be read in logcat and `floatingActionButton.getMenuButton()` is null.

###### Expected Result

`floatingActionButton.getMenuButton()` should not be null.

###### Actual Result

`floatingActionButton.getMenuButton()` is null.
And stack trace in logcat:
```gradle
W/System.err: android.content.res.Resources$NotFoundException: File res/drawable/ic_add_white_24dp.xml from drawable resource ID #0x7f080071
W/System.err:     at android.content.res.Resources.loadDrawable(Resources.java:2101)
W/System.err:     at android.content.res.TypedArray.getDrawable(TypedArray.java:602)
W/System.err:     at jahirfiquitiva.libs.fabsmenu.FABsMenu.init(FABsMenu.java:152)
W/System.err:     at jahirfiquitiva.libs.fabsmenu.FABsMenu.<init>(FABsMenu.java:106)
W/System.err:     at java.lang.reflect.Constructor.constructNative(Native Method)
W/System.err:     at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
W/System.err:     at android.view.LayoutInflater.createView(LayoutInflater.java:595)
W/System.err:     at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:697)
W/System.err:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:756)
W/System.err:     at android.view.LayoutInflater.parseInclude(LayoutInflater.java:840)
W/System.err:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
W/System.err:     at android.view.LayoutInflater.rInflate(LayoutInflater.java:759)
W/System.err:     at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
W/System.err:     at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
W/System.err:     at android.view.LayoutInflater.inflate(LayoutInflater.java:353)
W/System.err:     at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287)
W/System.err:     at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139)
W/System.err:     at com.amaze.filemanager.activities.MainActivity.onCreate(MainActivity.java:337)
W/System.err:     at android.app.Activity.performCreate(Activity.java:5231)
W/System.err:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
W/System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
W/System.err:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
W/System.err:     at android.app.ActivityThread.access$800(ActivityThread.java:135)
W/System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:102)
W/System.err:     at android.os.Looper.loop(Looper.java:136)
W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:5017)
W/System.err:     at java.lang.reflect.Method.invokeNative(Native Method)
W/System.err:     at java.lang.reflect.Method.invoke(Method.java:515)
W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
W/System.err:     at dalvik.system.NativeStart.main(Native Method)
W/System.err: Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #0: invalid drawable tag vector
W/System.err:     at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:933)
W/System.err:     at android.graphics.drawable.Drawable.createFromXml(Drawable.java:877)
W/System.err:     at android.content.res.Resources.loadDrawable(Resources.java:2097)
W/System.err: 	... 31 more
```

--- a/library/src/main/java/jahirfiquitiva/libs/fabsmenu/FABsMenu.java
+++ b/library/src/main/java/jahirfiquitiva/libs/fabsmenu/FABsMenu.java
@@ -40,6 +40,7 @@ import android.support.annotation.IntDef;
 import android.support.annotation.NonNull;
 import android.support.design.widget.CoordinatorLayout;
 import android.support.design.widget.FloatingActionButton;
+import android.support.graphics.drawable.VectorDrawableCompat;
 import android.support.v4.content.ContextCompat;
 import android.support.v7.widget.RecyclerView;
 import android.util.AttributeSet;
@@ -67,6 +68,7 @@ public class FABsMenu extends ViewGroup {
     public static final int EXPAND_RIGHT = 3;
     public static final int LABELS_ON_LEFT_SIDE = 0;
     public static final int LABELS_ON_RIGHT_SIDE = 1;
+    private static final String TAG = FABsMenu.class.getSimpleName();
     private static final float COLLAPSED_PLUS_ROTATION = 0f;
     private static final float EXPANDED_PLUS_ROTATION = 90f + 45f;
     private static Interpolator expandInterpolator = new OvershootInterpolator();
@@ -115,17 +117,16 @@ public class FABsMenu extends ViewGroup {
     }
 
     private void init(Context context, AttributeSet attributeSet) {
+        buttonSpacing = (int) DimensionUtils.convertDpToPixel(16, context);
+        labelsMargin = getResources().getDimensionPixelSize(R.dimen.fab_labels_margin);
+        labelsVerticalOffset = (int) DimensionUtils.convertDpToPixel(-1.5f, context);
+
+        touchDelegateGroup = new TouchDelegateGroup(this);
+        setTouchDelegate(touchDelegateGroup);
+
+        TypedArray attr = context.obtainStyledAttributes(attributeSet, R.styleable.FABsMenu,
+                0, 0);
         try {
-            buttonSpacing = (int) DimensionUtils.convertDpToPixel(16, context);
-            labelsMargin = getResources().getDimensionPixelSize(R.dimen.fab_labels_margin);
-            labelsVerticalOffset = (int) DimensionUtils.convertDpToPixel(-1.5f, context);
-
-            touchDelegateGroup = new TouchDelegateGroup(this);
-            setTouchDelegate(touchDelegateGroup);
-
-            TypedArray attr = context.obtainStyledAttributes(attributeSet, R.styleable.FABsMenu,
-                                                             0, 0);
-
             menuMargins = attr.getDimensionPixelSize(R.styleable.FABsMenu_fab_menuMargins, 0);
 
             menuTopMargin = attr.getDimensionPixelSize(R.styleable.FABsMenu_fab_menuTopMargin,
@@ -152,7 +153,10 @@ public class FABsMenu extends ViewGroup {
                                                         : (int) DimensionUtils
                                                                 .convertDpToPixel(16, context));
 
-            menuButtonIcon = attr.getDrawable(R.styleable.FABsMenu_fab_moreButtonPlusIcon);
+            int drawableId = attr.getResourceId(R.styleable.FABsMenu_fab_moreButtonPlusIcon, 0);
+            if(drawableId != 0){
+                menuButtonIcon = VectorDrawableCompat.create(getResources(), drawableId, context.getTheme());
+            }
 
             menuButtonColor = attr.getColor(R.styleable.FABsMenu_fab_moreButtonBackgroundColor,
                                             getColor(android.R.color.holo_blue_dark));
@@ -165,18 +169,20 @@ public class FABsMenu extends ViewGroup {
             expandDirection = attr.getInt(R.styleable.FABsMenu_fab_expandDirection, EXPAND_UP);
 
             labelsPosition = attr.getInt(R.styleable.FABsMenu_fab_labelsPosition,
-                                         isRtl() ? LABELS_ON_RIGHT_SIDE : LABELS_ON_LEFT_SIDE);
+                    isRtl() ? LABELS_ON_RIGHT_SIDE : LABELS_ON_LEFT_SIDE);
 
-            attr.recycle();
 
-            if (menuListener == null) {
-                setMenuListener(new FABsMenuListener() {
-                });
-            }
-            createMenuButton(context);
         } catch (Exception e) {
-            e.printStackTrace();
+            Log.w(TAG, "Failure reading attributes", e);
+        } finally {
+            attr.recycle();
         }
+
+        if (menuListener == null) {
+            setMenuListener(new FABsMenuListener() {
+            });
+        }
+        createMenuButton(context);
     }
 
     private boolean expandsHorizontally() {
@@ -235,7 +241,7 @@ public class FABsMenu extends ViewGroup {
     public void addButton(TitleFAB button, int index) throws IllegalArgumentException {
         if (buttonsCount >= 6)
             throw new IllegalArgumentException("A floating action buttons menu should have no " +
-                                                       "more than six options.");
+                    "more than six options.");
         addView(button, index);
         buttonsCount += 1;
         createLabels();
@@ -243,7 +249,7 @@ public class FABsMenu extends ViewGroup {
             show(false);
         }
         if (buttonsCount < 3)
-            Log.w("FABsMenu", "A floating action buttons menu should have at least three options");
+            Log.w(TAG, "A floating action buttons menu should have at least three options");
     }
 
     public void addButton(TitleFAB button) throws IllegalArgumentException {
@@ -284,7 +290,7 @@ public class FABsMenu extends ViewGroup {
             buttonsCount -= 1;
             if (buttonsCount <= 1) hide();
         } catch (Exception e) {
-            e.printStackTrace();
+            Log.e(TAG, "Failure removing Button", e);
         }
     }
 
@@ -896,7 +902,7 @@ public class FABsMenu extends ViewGroup {
             Drawable icon = Drawable.createFromStream(inputStream, uri.toString());
             if (icon != null) setMenuButtonIcon(icon);
         } catch (Exception e) {
-            e.printStackTrace();
+            Log.e(TAG, "Failure setting MenuButton icon", e);
         }
     }
 
@@ -907,7 +913,7 @@ public class FABsMenu extends ViewGroup {
             else throw new NullPointerException(
                     "The icon you try to assign to FABsMenu does not exist");
         } catch (Exception e) {
-            e.printStackTrace();
+            Log.e(TAG, "Failure setting MenuButton icon", e);
         }
     }
 
@@ -993,7 +999,7 @@ public class FABsMenu extends ViewGroup {
         try {
             setMenuButtonIcon(new BitmapDrawable(getResources(), bitmap));
         } catch (Exception e) {
-            e.printStackTrace();
+           Log.e(TAG, "Failure setting MenuButton icon", e);
         }
     }
 

--- a/library/src/main/java/jahirfiquitiva/libs/fabsmenu/FABsMenuLayout.java
+++ b/library/src/main/java/jahirfiquitiva/libs/fabsmenu/FABsMenuLayout.java
@@ -27,6 +27,7 @@ import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.support.design.widget.CoordinatorLayout;
 import android.util.AttributeSet;
+import android.util.Log;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.FrameLayout;
@@ -34,6 +35,7 @@ import android.widget.FrameLayout;
 @CoordinatorLayout.DefaultBehavior(FABSnackbarBehavior.class)
 public class FABsMenuLayout extends FrameLayout {
 
+    private static final String TAG = FABsMenuLayout.class.getSimpleName();
     @ColorInt
     private int overlayColor;
     private View overlayView;
@@ -56,25 +58,27 @@ public class FABsMenuLayout extends FrameLayout {
     }
 
     private void init(Context context, AttributeSet attrs) {
+        TypedArray attr = context.getTheme().obtainStyledAttributes(attrs,
+                R.styleable.FABsMenuLayout, 0,
+                0);
         try {
-            TypedArray a = context.getTheme().obtainStyledAttributes(attrs,
-                                                                     R.styleable.FABsMenuLayout, 0,
-                                                                     0);
-            overlayColor = a.getColor(R.styleable.FABsMenuLayout_fabs_menu_overlayColor,
+            overlayColor = attr.getColor(R.styleable.FABsMenuLayout_fabs_menu_overlayColor,
                                       Color.parseColor("#4d000000"));
-            clickableOverlay = a.getBoolean(R.styleable.FABsMenuLayout_fabs_menu_clickableOverlay,
+            clickableOverlay = attr.getBoolean(R.styleable.FABsMenuLayout_fabs_menu_clickableOverlay,
                                             true);
-            a.recycle();
-
-            overlayView = new View(context);
-            overlayView.setLayoutParams(new FrameLayout.LayoutParams(
-                    ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
-            overlayView.setBackgroundColor(overlayColor);
-            overlayView.setVisibility(View.GONE);
-            addView(overlayView);
         } catch (Exception e) {
-            e.printStackTrace();
+            Log.e(TAG, "Failure setting MenuButton icon", e);
+        }finally {
+            attr.recycle();
         }
+
+        overlayView = new View(context);
+        overlayView.setLayoutParams(new FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
+        overlayView.setBackgroundColor(overlayColor);
+        overlayView.setVisibility(View.GONE);
+        addView(overlayView);
+
     }
 
     @ColorInt

--- a/library/src/main/java/jahirfiquitiva/libs/fabsmenu/TitleFAB.java
+++ b/library/src/main/java/jahirfiquitiva/libs/fabsmenu/TitleFAB.java
@@ -30,6 +30,7 @@ import android.support.v4.view.ViewCompat;
 import android.support.v4.view.animation.FastOutLinearInInterpolator;
 import android.support.v4.view.animation.LinearOutSlowInInterpolator;
 import android.util.AttributeSet;
+import android.util.Log;
 import android.view.View;
 import android.view.animation.Interpolator;
 
@@ -51,6 +52,7 @@ public class TitleFAB extends FloatingActionButton {
     private static final int ANIM_STATE_NONE = 0;
     private static final int ANIM_STATE_HIDING = 1;
     private static final int ANIM_STATE_SHOWING = 2;
+    private static final String TAG = TitleFAB.class.getSimpleName();
 
     int mAnimState = ANIM_STATE_NONE;
 
@@ -80,26 +82,28 @@ public class TitleFAB extends FloatingActionButton {
     }
 
     void init(Context context, AttributeSet attributeSet) {
+        TypedArray attr =
+                context.obtainStyledAttributes(attributeSet, R.styleable.TitleFAB, 0, 0);
         try {
-            TypedArray attr =
-                    context.obtainStyledAttributes(attributeSet, R.styleable.TitleFAB, 0, 0);
             title = attr.getString(R.styleable.TitleFAB_fab_title);
             titleClickEnabled = attr.getBoolean(R.styleable.TitleFAB_fab_enableTitleClick, true);
             titleBackgroundColor = attr.getInt(R.styleable.TitleFAB_fab_title_backgroundColor,
-                                               ContextCompat
-                                                       .getColor(context, android.R.color.white));
+                    ContextCompat
+                            .getColor(context, android.R.color.white));
             titleTextColor = attr.getInt(R.styleable.TitleFAB_fab_title_textColor,
-                                         ContextCompat.getColor(context, android.R.color.black));
+                    ContextCompat.getColor(context, android.R.color.black));
             titleCornerRadius =
                     attr.getDimensionPixelSize(R.styleable.TitleFAB_fab_title_cornerRadius, -1);
             titleTextPadding =
                     attr.getDimensionPixelSize(R.styleable.TitleFAB_fab_title_textPadding,
-                                               (int) DimensionUtils.convertDpToPixel(8, context));
-            setSize(FloatingActionButton.SIZE_MINI);
+                            (int) DimensionUtils.convertDpToPixel(8, context));
+        } catch (Exception e) {
+            Log.w(TAG, "Failure reading attributes", e);
+        } finally {
             attr.recycle();
-            setOnClickListener(null);
-        } catch (Exception ignored) {
         }
+        setOnClickListener(null);
+        setSize(FloatingActionButton.SIZE_MINI);
     }
 
     @Override
@@ -207,7 +211,7 @@ public class TitleFAB extends FloatingActionButton {
         LabelView label = getLabelView();
         if (label != null && label.getContent() != null) {
             label.getContent().setPadding(titleTextPadding, titleTextPadding / 2, titleTextPadding,
-                                          titleTextPadding / 2);
+                    titleTextPadding / 2);
         }
     }
 

